lib/xenops_server.ml

```diff
@@ -3299,6 +3299,9 @@ module VM = struct
           import_metadata id md)
       ()
 
+  let import_metadata_async _ dbg s =
+    let id, md = parse_metadata s in
+    queue_operation dbg id (Atomic (VM_import_metadata (id, md)))
```

Comment:
I think you could potentially deduplicate multiple queued import metadata requests into a single one.
Updating `let should_keep` in `xenops_server.ml` should achieve that.

Reply:
Aha, that is useful. As long as it keep the last one in a list of consecutive `VM_import_metadata` ops, we don't care about the earlier ones. I'll look into that.

Reply:
Having briefly looked at this now, I think it should be possible to add `VM_import_metadata` to the `should_keep` function to coalesce multiple consecutive ones. However, we absolutely must keep the one that was added, and I wasn't able to work out yet how that corresponds to the "previous" items in the queue (which way around is it). I propose to leave this for now, as I don't want to mess it up, and the gains are minimal.